### PR TITLE
Bug fix on batch size

### PR DESCRIPTION
**Pull request by berose**
Fixed bug on batch size bigger than 1
- [x] Let the loss backpropagated after being calculated through the whole batch
- [x] Fixed the indentation bug for prune
- [x] Tested the generation endpoint with batch size bigger than 1

### CODE REVIEW

File: generation/configs/text.yaml [hunk 1]
```diff
@@ -35,9 +35,9 @@ lambda_zero123: 0
 # warmup rgb supervision for image-to-3d
 warmup_rgb_loss: True
 # training batch size per iter
-batch_size: 1
+batch_size: 4
```
Comment (loayei) [inline generation/configs/text.yaml:38]: why 4 not 5?

Comment (berose) [reply generation/configs/text.yaml:38]: for test purpose only
can ignore changes on this file
